syscalls.go

```diff
@@ -469,3 +471,34 @@ var haveBatchAPI = internal.FeatureTest("map batch api", "5.6", func() error {
 	}
 	return nil
 })
+
+var haveProbeReadKernel = internal.FeatureTest("bpf_probe_read_kernel", "5.5", func() error {
+	insns := asm.Instructions{
+		asm.Mov.Reg(asm.R1, asm.R10),
+		asm.Add.Imm(asm.R1, -8),
+		asm.Mov.Imm(asm.R2, 8),
+		asm.Mov.Imm(asm.R3, 0),
+		asm.FnProbeReadKernel.Call(),
+		asm.Return(),
+	}
+	buf := bytes.NewBuffer(make([]byte, 0, len(insns)*asm.InstructionSize))
+	err := insns.Marshal(buf, internal.NativeEndian)
+	if err != nil {
+		return err
+	}
+	bytecode := buf.Bytes()
+
+	attr := &bpfProgLoadAttr{
+		progType: Kprobe,
+		license:  internal.NewStringPointer("GPL"),
+	}
+	attr.instructions = internal.NewSlicePointer(bytecode)
+	attr.insCount = uint32(len(bytecode) / asm.InstructionSize)
```

Comment:
nit: this could be moved directly into the struct creation above.

Reply:
Updated.